### PR TITLE
Fix novaclient api

### PR DESCRIPTION
**Pull request by Akm0d**
### What does this PR do?
Port https://github.com/saltstack/salt/pull/49977 to master

### CODE REVIEW

**Comment by dwoz** [comment]
@Akm0d The linter is still failing here.

**Comment by Akm0d** [comment]
re-run full all

**Comment by Akm0d** [comment]
re-run full windows2016

**Comment by Akm0d** [comment]
re-run full windows2019

**Comment by Akm0d** [comment]
re-run full opensuse15

**Review by s0undt3ch — APPROVED** [review]

File: salt/utils/openstack/nova.py [hunk 1]
```diff
@@ -555,10 +570,10 @@ def volume_show(self, name):
             search_opts={'display_name': name},
         )
         volume = volumes[name]
-#        except Exception as esc:
-#            # volume doesn't exist
-#            log.error(esc.strerror)
-#            return {'name': name, 'status': 'deleted'}
+        #        except Exception as esc:
```
Comment (dwoz) [inline salt/utils/openstack/nova.py]: why not just remove this?

File: salt/utils/openstack/nova.py [hunk 2]
```diff
@@ -259,9 +270,11 @@ def __init__(
                            os_auth_plugin=os_auth_plugin,
                            **kwargs)
 
-    def _new_init(self, username, project_id, auth_url, region_name, password, os_auth_plugin, auth=None, verify=True, **kwargs):
+    def _new_init(self, username, project_id, auth_url, region_name, password, os_auth_plugin, auth=None, verify=True,
+                  **kwargs):
         if auth is None:
             auth = {}
+        verify = kwargs.get('verify', False)
```
Comment (dwoz) [inline salt/utils/openstack/nova.py]: This doesn't make sense, how would verify get in `kwargs` when it's defined as a keyword argument on line 273?

**Review by dwoz — CHANGES_REQUESTED** [review]

File: salt/utils/openstack/nova.py [hunk 3]
```diff
@@ -303,7 +316,9 @@ def _new_init(self, username, project_id, auth_url, region_name, password, os_au
         conn = client.Client(version=self.version, session=self.session, **self.client_kwargs)
         self.kwargs['auth_token'] = conn.client.session.get_token()
         identity_service_type = kwargs.get('identity_service_type', 'identity')
-        self.catalog = conn.client.session.get('/auth/catalog', endpoint_filter={'service_type': identity_service_type}).json().get('catalog', [])
+        self.catalog = conn.client.session.get('/auth/catalog',
+                                               endpoint_filter={'service_type': identity_service_type}).json().get(
```
Comment (dwoz) [inline salt/utils/openstack/nova.py]: This line is still really long, I suggest

```
   self.catalog = conn.client.session.get(
        '/auth/catalog',
        endpoint_filter={
            'service_type': identity_service_type
        }
    ).json().get('catalog', [])
```

**Comment by dwoz** [comment]
2019 windows tests are passing:

```
05:57:53         OK (total=9282, skipped=1213, passed=8069, failures=0, errors=0)
```

merging pr

**Pull request merged**